NBISweden/sda-cli
# Download files via htsget

**Related issue(s) and PR(s)**  
This PR closes #280.


**Description**
This adds a component called htsget to the sda-cli that allows to download (partial) files through the htsget server.

**How to test**
To test this, you need to run the _docker compose_ of `genomicDataInfra/starter-kit-storage-and-interfaces` with:
`docker compose -f docker-compose-demo.yml up -d` 
Then get the token as listed in that repo and insert it into the testing/s3cmd.conf file of the sda-cli repo with the name of `access_token=<token>` 
Last step for the dev environment is to run the _docker compose_ from `genomicDataInfra/starter-kit-htsget` with `docker compose up -d`
Then you can run the htsget sda-cli command with something like:
`./sda-cli htsget -config testing/s3cmd.conf -dataset DATASET0001 -filename htsnexus_test_NA12878 -reference 11 -host http://localhost:8088 -pubkey testing/c4gh.pub.pem -output ./result.c4gh --force-overwrite`, where `pubkey` is your own c4gh key pair provided to be used. 
This will download the file `result.c4gh`
You can then decrypt the file with:
`./sda-cli decrypt -key c4gh.sec.pem result.c4gh`
To check that it worked, check the file contents with samtools:
`samtools view result`
